### PR TITLE
Document Code and Implement Previous Iterator

### PR DESCRIPTION
**Pull request by appcypher**
## Summary

This PR implements the following **features** and **fixes**

- [x] Add documentation to API 
- [x] Implement Iterator design pattern for revision history
- [x] Make code more idiomatic
- [x] Add structure to code

## Test plan (required)

-  Run tests.
    
    ```bash
    cargo test --release
    ```

## Closing issues

Fixes https://github.com/WebNativeFileSystem/rs-skip-ratchet/issues/1

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 /target
+**/.DS_Store
```
Comment (expede) [inline .gitignore:2]: @appcypher Protip: you can add this to your system gitignore, and never have to deal with this in projects again! https://www.flother.is/til/global-gitignore/

Comment (appcypher) [reply .gitignore:2]: Handy

File: src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    }
+
+    // fast jump 256 values in one shot
+    let (ref fast, _) = Ratchet::zero(seed.into()).next_medium_epoch();
+    assert_ratchet_equal(slow, fast);
+}
+
+// #[test]
+// fn test_fuzz_ratchet {
+//     // TODO: research rust fuzz impl
```
Comment (matheus23) [inline src/tests.rs:57]: For my IBF stuff I've recently used `proptest`. The docs aren't *amazing*, but it's good to use, and actively updated (unlike quickcheck). Unfortunately you have to use a `proptest!` macro, and I always find the domain-specific macros the hardest thing to learn & you sometimes lose IDE intellisense, but once I got what it's doing it was fine :)

**Review by matheus23 — APPROVED** [review]
Nice work! :tada:

File: src/tests.rs (hunk 2 above)
Comment (appcypher) [reply src/tests.rs:57]: Oh. The comment is from the original code. I'm leaving it there cos we do need fuzz testing. I'm checking out proptest.

**Pull request merged**